### PR TITLE
feat: CSAT for all channels

### LINKED ISSUE #2653: CSAT for every channel (backend)
Right now, CSAT surveys are supported only in the web-widget channel. To support CSAT surveys in all channels, we would start by sending a link once the conversation is resolved.

Once the conversation is resolved, the system should create a message with input_csat type. In each send_on_service listener, there should be an additional check to alter the content to send the link for the CSAT survey.

URL format: `<installation-url>/app/responses/<response-uuid>`

**Todo**:

- [x] #2671
- [ ] Update API to provide rating and response. Save the response in both the message object as well as the CSAT response object.
- [ ] Behaviour: 
   - [ ] Lock interaction based on the day: If the CSAT is a message sent more than 14 days before, then throw 404.
   - [ ] If already submitted, add an additional flag on the response to show that the message is submitted.

**Notes**

- Shouldn't work for twitter

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes #2653

### CODE REVIEW

File: app/models/message.rb [hunk 1]
```diff
@@ -130,6 +130,13 @@ def webhook_data
     }
   end
 
+  def content
+    # move this to a presenter
+    return self[:content] if !input_csat? || !inbox.web_widget?
+
+    "Please rate this conversation , #{ENV['FRONTEND_URL']}/survey/responses/#{conversation.uuid}"
```
Comment (sojan-official) [inline app/models/message.rb:137]: Lets add translations for this

**Review by sojan-official — COMMENTED** [review]
The code looks good to me. Let's do a round of testing for the behavior on all channels and get this merged. 

- shouldn't get triggered for tweet conversations

**Comment by sojan-official** [comment]
@pranavrajs  @muhsin-k  we might need to revisit whether we should start new conversations for Facebook and Twitter on resolution.

**Comment by sojan-official** [comment]
## email 

Screenshot 2021-08-21 at 12 08 47 AM

## response page

Screenshot 2021-08-21 at 12 08 38 AM

**Review by sojan-official — DISMISSED** [review]

**Comment by sojan-official** [comment]
@nithindavid  could you take a look at the front end code, if it's all good let's get this merged?

File: app/javascript/survey/views/Response.vue [hunk 2]
```diff
@@ -33,26 +47,138 @@
...
+    isRatingSubmitted() {
+      return this.surveyDetails && this.surveyDetails.rating;
+    },
+    isFeedbackSubmitted() {
+      return this.surveyDetails && this.surveyDetails.feedback_message;
+    },
+    isButtonDisabled() {
+      return !(this.selectedRating && this.feedback);
+    },
+    enableBanner() {
```
Comment (nithindavid) [inline app/javascript/survey/views/Response.vue:101]: Would be nice to see `isSomething` pattern being followed like other functions to keep it consisent.

File: app/javascript/survey/components/Feedback.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <label class="text-base font-medium text-black-800">
+      {{ $t('SURVEY.FEEDBACK.LABEL') }}
+    </label>
+    <text-area
+      v-model="feedback"
+      class="my-5"
+      :placeholder="$t('SURVEY.FEEDBACK.PLACEHOLDER')"
+    />
+    <custom-button
+      class="flex items-center font-medium float-right"
```
Comment (nithindavid) [inline app/javascript/survey/components/Feedback.vue:12]: We could avoid float classes as we have flex styles. We could wrap it in a div to float it right.

**Review by nithindavid — CHANGES_REQUESTED** [review]
@muhsin-k Nice work :D I have left some minor comments

**Review by nithindavid — APPROVED** [review]

**Pull request merged**